### ISSUE TITLE
Add support for the image-based mode of the icon macro

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor-pdf#1711]
Unlike other converters, when the `icons` document attribute is set to empty string or `image`, local images are not used for the icon macro. Instead of looking for a local image, the alt text is shown. A local image should be used instead. Construct the path by concatenating the value of the 'iconsdir' attribute, the icon name, and the value of the 'icontype' attribute (which defaults to 'png').

**Comment by mojavelinux** [comment]
This has actually worked for a long time. However, the converter didn't fully implement the logic for embedding an image, so certain features were not available, such as a remote URL. That was fixed by d2ca65eabdba070ede32d16ba1b492c2de1c3c04.
